ivadomed/loader/mri3d_subvolume_segmentation_dataset.py

```diff
@@ -148,86 +148,112 @@ def __len__(self):
         """Return the dataset size. The number of subvolumes."""
         return len(self.indexes)
 
-    def __getitem__(self, index):
+    def __getitem__(self, subvolume_index: int):
         """Return the specific index pair subvolume (input, ground truth).
 
         Args:
-            index (int): Subvolume index.
+            subvolume_index (int): Subvolume index.
         """
 
         # copy.deepcopy is used to have different coordinates for reconstruction for a given handler,
         # to allow a different rater at each iteration of training, and to clean transforms params from previous
         # transforms i.e. remove params from previous iterations so that the coming transforms are different
-        coord = self.indexes[index]
 
-        tuple_seg_roi_pair: tuple = self.handlers[coord['handler_index']]
+        # Get the tuple that defines the boundaries for the subsample
+        coord: dict = self.indexes[subvolume_index]
+        x_min = coord.get(SegmentationDatasetKW.X_MIN)
+        x_max = coord.get(SegmentationDatasetKW.X_MAX)
+        y_min = coord.get(SegmentationDatasetKW.Y_MIN)
+        y_max = coord.get(SegmentationDatasetKW.Y_MAX)
+        z_min = coord.get(SegmentationDatasetKW.Z_MIN)
+        z_max = coord.get(SegmentationDatasetKW.Z_MAX)
+
+        # Obtain tuple reference to the pairs of file references
+        tuple_seg_roi_pair: tuple = self.handlers[coord.get(SegmentationDatasetKW.HANDLER_INDEX)]
```

Comment:
The use of the dictionary is a nice addition. We should also use it in the [function](https://github.com/ivadomed/ivadomed/blob/ed964e7233d5f56d0597951669dedc71a68e27ba/ivadomed/loader/mri3d_subvolume_segmentation_dataset.py#L139-L145) that create the indices.

Reply:
Yeah, refactoring and keywords extractions are going to be in separate PRs. Trying to limit change scopes within the same functions as main purpose.